### ISSUE TITLE
Any character needing a dead key cannot by typed in tox's edit window: ô, ï, ...

### ISSUE THREAD

**Issue by FabrizioMarana** [qTox/qTox#2564]
If you copy-paste the following characters into the tox edit window they get displayed correctly:
äëïöü, âêîôû çñã
However, when typed from a keyboard, they do not.

Application: qtox-unity
uname --kernel-release&&lsb_release --code --short
3.13.0-68-generic
trusty

To reproduce: Choose "English (US, alternative international keyboard)" and type ` followed by i to get ì, ' followed by i to get í, " and e to get ë etc.

So basically "dead keys" do not work... (tried not only US international but German, French keyboard layouts as well: all the same problem)

**Comment by agilob** [comment]
It's a Qt issue:
https://bbs.archlinux.org/viewtopic.php?id=133210
https://bugreports.qt.io/browse/QTBUG-28183
http://www.qtcentre.org/threads/37106-Qt-on-Linux-and-dead-keys
http://ubuntuforums.org/showthread.php?t=2135167
https://bugs.kde.org/show_bug.cgi?id=118862
https://bugs.kde.org/show_bug.cgi?id=318151

If someone has account on qt bugs system, could you report this issue?

**Comment by FabrizioMarana** [comment]
I'm willing to do the effort of filing another bug as QTBUG-28183 seems to be closed, but I would need to have some technical info as the only thing I know about QT is that it's a library.

So which version do you use, what's the exact system call or where can I find this kind of information?  Just to give you some reference: I used to be a C developer a long time ago, but have never done any FLOSS development (except for a few bash scripts included in BackInTime)

**Comment by agilob** [comment]
I'll fill the bugreport today than. We got this issue reported twice already, one for Qt5.4 and now for 5.5.1, no one reported it before when we were on Qt5.1 or 5.2 for a long time.

**Comment by TheNain38** [comment]
@FabrizioMarana The Qt version used is in the about window of qTox

**Comment by FabrizioMarana** [comment]
@agilob : thanks, please post a link to the QT bug report in here as well, so I can rally some more people on the qt site as "I'm affected too"...  **;-)**
@TheNain38 : Oops, found 5.5.1 in the lower right of the panel of the about window...  (Missed that at first glance)

**Comment by agilob** [comment]
[Reported](https://bugreports.qt.io/browse/QTBUG-49438)

**Comment by ghost** [comment]
Was #2158 in some way related?

**Comment by agilob** [comment]
Oh, yes, thanks for updating. They marked is as critical so, let's hope it will be fixed soon o/

**Comment by yurivict** [comment]
Why input methods work with other Qt5 apps, but not with qTox?

**Comment by agilob** [comment]
Used qt version.

**Comment by yurivict** [comment]
I am not sure what this means "qt version", and if this is even the same problem.

I use ibus to switch layouts. When I type something in other Qt-5.4 apps layout works, but in qTox (also Qt-5.4) layout doesn't work.

**Comment by Nurmagoz** [comment]
how long should take this topic to solve the issue? i think the subject took long enough for solving, from Nov till now what more?!

**Comment by ghost** [comment]
@TNTBOMBOM this is the place you were looking for that comment:
https://bugreports.qt.io/browse/QTBUG-49438

**Comment by tux3** [comment]
We have so many duplicates I'm just going to add a new label for >= 5...

**Comment by ghost** [comment]
This is working for me now with Qt 5.5.1. Could the error be somewhere else? I don't know which info could be relevant, please ask any if needed.

To be precise:
- äëïöüâêîôûçñã and the like can be typed and pasted (UK keyboard)
- Japanese (Kana keyboard) and Arabic (azerty keyboard) can be typed and pasted (#3013, #2857 etc)
- е́ (Ukrainian keyboard), can't be typed nor pasted (#2158)

---

qTox a48c4225938834db0d4406cff54a07378e9636f4
Debian stretch
Cinnamon 2.8.6
Qt 5.5.1
Linux 4.3.0-1-amd64 x86_64

**Comment by ghost** [comment]
I have tested this now in an Ubuntu 16.04 Beta 2 64-bit VM and I find that I get all the same results as @kehugter (I am not able to paste that letter, but I can't find it on they keyboard so can't test if I can type it...). Though there is an issue with the Arabic, though when I can copy the message once it has been sent and it displays fine, I only see a blank space where the Arabic letters should be in qtox (in the recorder of what has been said, not in the message composing space). Is this thing with the Arabic a separate issue that I should file a report on or does this report cover it?

**Comment by Nurmagoz** [comment]
btw i figured out how u can type a full text in arabic and it can appear normally , but it is not sufficient to be used. but i hope it will help in solving the issue.

if u want ur typing to be appeared, try typing 40 letters (even if they similar) then after that type ur sentences or what ever u like, it will be like this:- 

![whonix3](https://cloud.githubusercontent.com/assets/11895339/14237516/753713d6-fa14-11e5-941c-8493140fc0a2.png)

so the code is:- 

> اااااااااااااااااااااااااااااااااااااااا

change the language to arabic , then press 40 time any character u like (i pressed on ا which is the H letter in english keyboard) and then type whatever u like and text going to show up as like in the photo. 

@tux3 @agilob @zetok  , how strange is that?

**Comment by Aerolivier** [comment]
This issue also applies to the Compose Key (on my machine I can press Alt Gr + ' + e and get é).

I know for certain in an earlier version this was working perfectly fine (though that may be quite a long time ago; I didn't think to look on the issue tracker until now)

**Comment by zetok** [comment]
Hmm, I can type `é` using compose key (`alt gr + ; + e`) on edc35ba9dfce0eae1557016933147c926f04a22d with Qt 5.5.1. Same goes for other characters.

Still works also after setting keyboard to use `en_US` layout with

```
setxkbmap -model pc105 -layout "en_US" -option "compose:ralt"
```

Do you have in `/usr/share/X11/locale/` `en_US`, or whatever else layout that you're using?

**Comment by Aerolivier** [comment]
Ah, my apologies!

I actually just cloned the repo and compiled (I was going to look at adding a feature..)
and Compose Key does indeed work in 1c45ca3 with Qt version: 5.5.1.

Edit: Previous edit about it not working in 'run' mode seems untrue now.

**Comment by zetok** [comment]
Oh, I see. Can anyone still reproduce this issue on newest qTox?

**Comment by ghost** [comment]
@zetok, I am.

**Comment by zetok** [comment]
@SeventhSonOfASeventhSon could you provide needed info? I.e. from https://bugreports.qt.io/browse/QTBUG-49438?focusedCommentId=313228&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-313228 :

> Looking at https://bugreports.qt.io/browse/QTBUG-28183 and https://bugreports.qt.io/browse/QTBUG-32461, I’m wondering if the absence of /usr/share/X11/locale/fr_FR.UTF-8/Compose could play a role (fr_FR.UTF-8 being the used locale in my case).

**Comment by ghost** [comment]
@zetok, Sorry, I looked at that, but still not sure exactly what information you need... Please be more specific, sorry if I am missing something obvious.

**Comment by zetok** [comment]
@SeventhSonOfASeventhSon 
- either `tree /usr/share/X11/locale/` / `ls -lR /usr/share/X11/locale/` 
- keyboard layout(s) you tried with.

**Comment by ghost** [comment]
@zetok 

I have tried it with the United Kingdom, the Ukrainian, the Japanese, and the Arabic keyboard layout so far. And here is the requested `tree` output:

> /usr/share/X11/locale/
> ├── am_ET.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── armscii-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── C
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── compose.dir
> ├── cs_CZ.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── el_GR.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── en_US.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── fi_FI.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── georgian-academy
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── georgian-ps
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── ibm-cp1133
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iscii-dev
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── isiri-3342
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-1
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-10
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-11
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-13
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-14
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-15
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-2
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-3
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-4
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-5
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-6
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-7
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-9
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── iso8859-9e
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── ja
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── ja.JIS
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── ja_JP.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── ja.SJIS
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── km_KH.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── ko
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── koi8-c
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── koi8-r
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── koi8-u
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── ko_KR.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── locale.alias
> ├── locale.dir
> ├── microsoft-cp1251
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── microsoft-cp1255
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── microsoft-cp1256
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── mulelao-1
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── nokhchi-1
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── pt_BR.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── pt_PT.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── ru_RU.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── sr_CS.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── tatar-cyr
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── th_TH
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── th_TH.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── tscii-0
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── vi_VN.tcvn
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── vi_VN.viscii
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── zh_CN
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── zh_CN.gb18030
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── zh_CN.gbk
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── zh_CN.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── zh_HK.big5
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── zh_HK.big5hkscs
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── zh_HK.UTF-8
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── zh_TW
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> ├── zh_TW.big5
> │   ├── Compose
> │   ├── XI18N_OBJS
> │   └── XLC_LOCALE
> └── zh_TW.UTF-8
>     ├── Compose
>     ├── XI18N_OBJS
>     └── XLC_LOCALE
> 
> 62 directories, 189 files

**Comment by zetok** [comment]
Thanks. Could you try with different keyboard layout, e.g. `en_US` ?

**Comment by ghost** [comment]
@zetok, Ok, tried that, unable to type anything requiring dead keys though I could with the UK one. I just get capital letters.

**Comment by nettlebay** [comment]
Me too.. It's a little different for me: I use a QWERTY **portuguese** keyboard but my language is french. I never had problems until today with this keyboard & other apps. Note there is no accents problem on my other computer with a french AZERTY keyboard.
My main computer has Ubuntu Mate 16.04 and my other one: LinuxLite 14.04 (XFCE)

**Comment by Nurmagoz** [comment]
UPDATE:- 

now if i want to write in arabic keyboard , i just type one english letter for e.g:- 

`f اهلا وهسلا`

or 

`k شكرا`

then it will fully show up.

**Comment by nettlebay** [comment]
No problem with accentued letters in Utox

**Comment by anthonybilinski** [comment]
It looks like this should be fixed in Qt 5.11: https://bugreports.qt.io/browse/QTBUG-56452
Linux systems will use whatever distro version is provided, for Ubuntu 18.04 at least that's 5.11.1 so should include the fix.

Could someone please verify that it's now fixed when running with Qt >= 5.11.0?

**Comment by anthonybilinski** [comment]
Closing based on no feedback, please comment if it's still observed with newer Qt.
